packages/server/app/Models/User.ts

```diff
@@ -160,9 +170,11 @@ export default class User extends BaseModel {
         {
             token,
             tokenType,
+            revoke = false,
         }: {
             token: string;
             tokenType: TokenTypeName;
+            revoke?: boolean;
```

Comment:
Question: I'm not sure to understand what this method job's doing ? It's not only about reading value but also mutating it ? Maybe should we create a sperated method `RevokeLastToken` that takes a tokenType ?

I mean that it's not because you achieve to retrieve a valid token that it will be "used" which mean needs to be revoked on the fly. Something could fail during the process we might want to wait the operation end before revoking a related token ?

Reply:
I can not agree more with you! I really went to the simplest solution.
The reason is that `checkToken` method does not return a `Token` but a boolean telling whether a token exists. We should replace this function with another one called like `getToken`, that actually returns a `Token`, to be able to revoke it **simply** afterwards.

If we keep `checkToken` function, we should create another function such as `revokeToken`, which would take a plain token as parameter, and that would have to iterate over all active tokens and compare their hashed value with the plain token.

Reply:
I think that a function that revokes _the last token_ could revoke another token than the one we want. We should better decide exactly which token we want to revoke.

Reply:
I see !
> I think that a function that revokes the last token could revoke another token than the one we want. We should better decide exactly which token we want to revoke.

Indeed there could be some race conditions :laughing: